### PR TITLE
Python: Add TestOnlyCommissionableDataProvider to pychip_server_native_init()

### LINKED ISSUE #21969: Core dump running Python Lighting example on Raspberry Pi 4
#### Problem

I'm trying to run the [Python based lighting example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app/python) on a Raspberry Pi 4 running Ubuntu 22.04 but get a core dump:

```
(python_env) root@ubuntu:/home/ubuntu/connectedhomeip-python/examples/lighting-app/python# python lighting.py 
[1660745048.570696][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_kvs-3HkuYN)
[1660745048.574383][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_kvs)
[1660745048.579531][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-HPFg6a)
[1660745048.579798][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.580077][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-8COg8H)
[1660745048.580246][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.580495][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-x0oe1E)
[1660745048.580656][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.580986][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-Dhxgk3)
[1660745048.581580][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.581647][79754:79754] CHIP:DL: NVS set: chip-factory/unique-id = "6E43A2356AA3FE56"
[1660745048.581842][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-C2Dly8)
[1660745048.582491][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.582559][79754:79754] CHIP:DL: NVS set: chip-factory/vendor-id = 65521 (0xFFF1)
[1660745048.582744][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-IhvBPE)
[1660745048.583263][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_factory.ini)
[1660745048.583328][79754:79754] CHIP:DL: NVS set: chip-factory/product-id = 32769 (0x8001)
[1660745048.583542][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-T9ukQV)
[1660745048.584009][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.584073][79754:79754] CHIP:DL: NVS set: chip-counters/reboot-count = 1 (0x1)
[1660745048.584246][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-qhIzBw)
[1660745048.584833][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.584897][79754:79754] CHIP:DL: NVS set: chip-counters/total-operational-hours = 0 (0x0)
[1660745048.585075][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-BbysPT)
[1660745048.585869][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1660745048.585931][79754:79754] CHIP:DL: NVS set: chip-counters/boot-reason = 0 (0x0)
[1660745048.586104][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-iaRuzO)
[1660745048.586599][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.586659][79754:79754] CHIP:DL: NVS set: chip-config/regulatory-location = 0 (0x0)
[1660745048.586803][79754:79754] CHIP:DL: writing settings to file (/tmp/chip_config.ini-KkBBCZ)
[1660745048.587847][79754:79754] CHIP:DL: renamed tmp file to file (/tmp/chip_config.ini)
[1660745048.587899][79754:79754] CHIP:DL: NVS set: chip-config/location-capability = 2 (0x2)
[1660745048.588413][79754:79754] CHIP:DL: Got Ethernet interface: eth0
[1660745048.588725][79754:79754] CHIP:DL: Found the primary Ethernet interface:eth0
[1660745048.589077][79754:79754] CHIP:DL: Got WiFi interface: wlan0
[1660745048.589143][79754:79754] CHIP:DL: Failed to reset WiFi statistic counts
[1660745048.589181][79754:79754] CHIP:DL: Device Configuration:
[1660745048.589255][79754:79754] CHIP:DL:   Serial Number: TEST_SN
[1660745048.589310][79754:79754] CHIP:DL:   Vendor Id: 65521 (0xFFF1)
[1660745048.589347][79754:79754] CHIP:DL:   Product Id: 32769 (0x8001)
[1660745048.589379][79754:79754] CHIP:DL:   Hardware Version: 0
Aborted (core dumped)
```

I had to apply the follow patch to get in building (will make a pull request for this at a later date):

```diff
diff --git a/scripts/build_python_device.sh b/scripts/build_python_device.sh
index 30779c6ab..894ea1bad 100755
--- a/scripts/build_python_device.sh
+++ b/scripts/build_python_device.sh
@@ -104,7 +104,7 @@ gn --root="$CHIP_ROOT" gen "$OUTPUT_ROOT" --args="chip_detail_logging=$chip_deta
 if [ "$enable_pybindings" == true ]; then
     ninja -v -C "$OUTPUT_ROOT" pycontroller
 else
-    ninja -v -C "$OUTPUT_ROOT" chip-library
+    ninja -v -C "$OUTPUT_ROOT" chip-core
 fi
 
 # Create a virtual environment that has access to the built python tools
@@ -115,7 +115,7 @@ virtualenv --clear "$ENVIRONMENT_ROOT"
 if [ "$enable_pybindings" == true ]; then
     WHEEL=("$OUTPUT_ROOT"/pybindings/pycontroller/pychip-*.whl)
 else
-    WHEEL=("$OUTPUT_ROOT"/controller/python/chip_library*.whl)
+    WHEEL=("$OUTPUT_ROOT"/controller/python/chip_core*.whl)
 fi
```

 #### Proposed Solution

N/A

### PR DESCRIPTION
**Pull request by sandeepmistry**
#### Issue Being Resolved
* Fixes #21969

#### Change overview
This change adds a `LinuxCommissionableDataProvider` to the Python chip server similar to [`examples/platform/linux/AppMain.cpp`](https://github.com/project-chip/connectedhomeip/blob/master/examples/platform/linux/AppMain.cpp#L181-L183) to prevent issue #21969. It also includes the changes from PR #22494 as they are needed to build.

Note: There is most likely a more eloquent platform independent way to do this, but wanted to get the discussion going.

cc/ @mrjerryjohns

### CODE REVIEW

File: src/controller/python/BUILD.gn [hunk 1]
```diff
@@ -73,10 +73,23 @@ shared_library("ChipDeviceCtrl") {
...
+      include_dirs += [ "${chip_root}/examples/platform/linux" ]
+
+      sources += [
+        "${chip_root}/examples/platform/linux/CommissionableInit.cpp",
+        "${chip_root}/examples/platform/linux/CommissionableInit.h",
+        "${chip_root}/examples/platform/linux/LinuxCommissionableDataProvider.cpp",
+        "${chip_root}/examples/platform/linux/LinuxCommissionableDataProvider.h",
+        "${chip_root}/examples/platform/linux/Options.cpp",
+        "${chip_root}/examples/platform/linux/testing/CustomCSRResponseOperationalKeyStore.cpp",
+        "${chip_root}/examples/platform/linux/testing/CustomCSRResponseOperationalKeyStore.h",
```
Comment (tcarmelveilleux) [inline src/controller/python/BUILD.gn:88]: Why this?

File: src/controller/python/chip/server/ServerInit.cpp [hunk 2]
```diff
@@ -138,6 +147,17 @@ void pychip_server_native_init()
         ChipLogError(DeviceLayer, "Failed to initialize CHIP stack: platform init failed: %s", chip::ErrorStr(err));
     }
 
+#if CHIP_DEVICE_LAYER_TARGET_LINUX
+    err = chip::examples::InitCommissionableDataProvider(gCommissionableDataProvider, LinuxDeviceOptions::GetInstance());
```
Comment (tcarmelveilleux) [inline src/controller/python/chip/server/ServerInit.cpp:151]: How are command line options parsed here? How do we ensure no clash with the Python command line options?

Comment (mrjerryjohns) [reply src/controller/python/chip/server/ServerInit.cpp:151]: Using `LinuxDeviceOptions` only works for C++ based example applications. It should not be used here.

I'd recommend this instead:

```
    static chip::DeviceLayer::TestOnlyCommissionableDataProvider TestOnlyCommissionableDataProvider;
    chip::DeviceLayer::SetCommissionableDataProvider(&TestOnlyCommissionableDataProvider);
```

Comment (sandeepmistry) [reply src/controller/python/chip/server/ServerInit.cpp]: @mrjerryjohns thank for the recommendation, I've updated in https://github.com/project-chip/connectedhomeip/pull/22498/commits/5de23efe466e49b7cb67f9139c923b6b71dffcee.

**Comment by sandeepmistry** [comment]
@mrjerryjohns this should be good to go now, with the changes in https://github.com/project-chip/connectedhomeip/pull/22498/commits/5f3f72cc308f840c2ab1e054ff8764e44b5dec69. The Python Lighting app example is now functional.

**Review by mrjerryjohns — APPROVED** [review]

**Pull request merged**